internal/oci/runtime_oci.go

```diff
@@ -285,13 +286,7 @@ func (r *runtimeOCI) ExecSyncContainer(ctx context.Context, c *Container, comman
 	}
 	defer os.RemoveAll(processFile)
 
-	pidDir, err := ioutil.TempDir("", "pidfile")
-	if err != nil {
-		return nil, err
-	}
-	defer os.RemoveAll(pidDir)
-
-	pidFile := filepath.Join(pidDir, c.id)
+	pidFile := filepath.Join(r.execNotifier.Directory(), c.id+uuid.New().String())
```

Comment:
What about using `ioutil.TempFile` together with the `r.execNotifier.Directory`?
```suggestion
	pidFile, err := ioutil.TempFile(r.execNotifier.Directory(), "exec-")
```

Reply:
ioutil.TempFile creates the file, which interacts strangely with the inotify code. I think it's better staying as is